OMSEventWriter.cpp

```diff
@@ -147,14 +147,18 @@ void OMSEventWriter::add_string(const std::string& value)
 
 void OMSEventWriter::add_string_field(const std::string& name, const std::string& value)
 {
-    _writer.Key(name.c_str(), name.size(), true);
-    _writer.String(value.c_str(), value.size(), true);
+    if (!_config.FilterFieldNameSet.count(name)) {
```

Comment:
Add this logic to add_int32_field and add_int64_field.

Reply:
Good point!